app/controllers/myott/mycommodities_controller.rb

```diff
@@ -0,0 +1,161 @@
+module Myott
+  class MycommoditiesController < MyottController
+    before_action :authenticate, only: %i[new create index]
+    before_action :initialize_subscription_data
+
+    require 'csv'
+    require 'roo'
+
+    VALID_FILE_TYPES = [
+      'text/csv',
+      'application/vnd.ms-excel',
+      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    ].freeze
+
+    def new
+      Rails.cache.delete(session[:subscription_key]) if session[:subscription_key].present?
+      Rails.cache.delete(session[:subscription_meta_key]) if session[:subscription_meta_key].present?
+      session.delete(:subscription_key)
+      session.delete(:subscription_meta_key)
+    end
+
+    def active
+      get_subscription_targets 'active'
+    end
+
+    def expired
+      get_subscription_targets 'expired'
+    end
+
+    def invalid
+      get_subscription_targets 'invalid'
+    end
+
+    def index
+      meta = Rails.cache.read(session[:subscription_meta_key]) if session[:subscription_meta_key].present?
```

Comment:
I'm finding the caching hard to follow. My questions would be:
1. Is it necessary, is there any improvement we can make to the backend
2. If yes, then can it be moved to the model or a service to keep the controller cleaner?